### ISSUE TITLE
How to train with multiple optimizers

### ISSUE THREAD

**Issue by X-funbean** [JDAI-CV/fast-reid#371]
## ❓ How to do something using fastreid
Hi, I'd like to implement a network which is showed below

![image](https://user-images.githubusercontent.com/24663255/103165597-f0d6e500-4854-11eb-9a4e-92eaf0543ffd.png)

It has 3 independent optimizers, each takes charge of partial/whole model. The training process is listed as follows
![image](https://user-images.githubusercontent.com/24663255/103165666-b0c43200-4855-11eb-82d2-8829f07771a6.png)
![image](https://user-images.githubusercontent.com/24663255/103165675-bde12100-4855-11eb-8db3-6553bd2e9141.png)

I can't find clues to implement multiple independent optimizers from the sample projects, any idea on how to do this?

**Comment by L1aoXingyu** [comment]
Actually, I think there are just 3 different gradients, not optimizers. You can just implement this by defining 3 loss function and `loss.backward()` can get the sum of the 3 different gradients.
